src/index/coinstatsindex.cpp

```diff
@@ -481,5 +486,5 @@ bool CoinStatsIndex::ReverseBlock(const CBlock& block, const CBlockIndex* pindex
     Assert(m_total_unspendables_scripts == read_out.second.total_unspendables_scripts);
     Assert(m_total_unspendables_unclaimed_rewards == read_out.second.total_unspendables_unclaimed_rewards);
 
-    return m_db->Write(DB_MUHASH, m_muhash);
```

Comment:
Seems like the only problem is this lone line?

Reply:
If I replace only that line with `true` the reorg test in `feature_coinstatsindex.py` still passes. But not sure if that helps with the original issue.

Reply:
>  if the node gets shut down during the reorg,

I guess this is very non trivial to test, but maybe add a comment?

Reply:
I added a comment at the end of `CoinStatsIndex::WriteBlock` (as suggested by ryanofsky below). I think that duplicating this comment in `CoinStatsIndex::ReverseBlock` is not required, since this function doesn't write anything else to disk there wouldn't really be the expectation that for it to write `DB_MUHASH` here.